### PR TITLE
Fix grammar and writing style for some of the Router documentation

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @chunhtai

File: packages/flutter/lib/src/widgets/router.dart [hunk 1]
```diff
@@ -297,11 +295,11 @@ class Router<T> extends StatefulWidget {
...
   ///
-  /// Use this method when you need to access the delegates in the [Router].
-  /// For example, you need to access the [backButtonDispatcher] of the parent
-  /// router to create a [ChildBackButtonDispatcher] for a nested router.
-  /// Another use case may be updating the value in [routeInformationProvider]
-  /// to navigate to a new route.
+  /// This method provides access to the delegates in the [Router]. For example,
+  /// this can be used to access the [backButtonDispatcher] of the parent router
+  /// when creating a [ChildBackButtonDispatcher] for a nested [Router].
+  /// Similarly, it might be used to cause the [routeInformationProvider] to
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/router.dart:301]: Oops, this is a false statement, the routeInformationProvider does not guarantee to have value setter. I originally added this statement when I added value setter in PlatformRouteInformationProvider, and it was removed as a part of addressing pr reviews, I forgot to remove the comment here.

**Review by chunhtai — APPROVED** [review]
Thanks for updating the doc. LGTM.

File: packages/flutter/lib/src/widgets/router.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/widgets/router.dart:301]: Sorry can you clarify exactly what I should change? I didn't quite follow your description.

Comment (chunhtai) [reply packages/flutter/lib/src/widgets/router.dart:301]: I mean we should remove line 301 and line 302.

basically there is not API in routeInformationProvider to change the route by default.

Comment (Hixie) [reply packages/flutter/lib/src/widgets/router.dart:301]: got it. done.

File: packages/flutter/lib/src/widgets/router.dart [hunk 2]
```diff
@@ -64,31 +67,29 @@ class RouteInformation {
 /// button), parses route information into data of type `T`, and then converts
 /// that data into [Page] objects that it passes to a [Navigator].
 ///
-/// Additionally, every single part of that previous sentence can be overridden
-/// and configured as desired.
+/// Each part of this process can be overridden and configured as desired.
```
Comment (yringler) [inline packages/flutter/lib/src/widgets/router.dart:70]: Yes, this is much more professional, but I did really  love the old version. It looked like it was written by a really excited dev after a late night of hard work and a lot of coffee.

Comment (Hixie) [reply packages/flutter/lib/src/widgets/router.dart:70]: I suspect it was originally written by me when I was designing the original Router API. :-)

**Pull request merged**